src/vea/vea_free.c

```diff
@@ -32,22 +32,130 @@ enum vea_free_type {
 	VEA_TYPE_PERSIST,
 };
 
+static d_list_t *
+blkcnt_to_lru(struct vea_free_class *vfc, uint32_t blkcnt)
+{
+	int idx;
+
+	D_ASSERTF(blkcnt <= vfc->vfc_sizes[0], "%u, %u\n",
+		  blkcnt, vfc->vfc_sizes[0]);
+	D_ASSERT(vfc->vfc_lru_cnt > 0);
+
+	for (idx = 0; idx < (vfc->vfc_lru_cnt - 1); idx++) {
+		if (blkcnt > vfc->vfc_sizes[idx + 1])
+			break;
+	}
+
+	return &vfc->vfc_lrus[idx];
+}
+
+void
+free_class_remove(struct vea_free_class *vfc, struct vea_entry *entry)
+{
+	if (entry->ve_in_heap) {
+		D_ASSERTF(entry->ve_ext.vfe_blk_cnt > vfc->vfc_large_thresh,
+			  "%u <= %u", entry->ve_ext.vfe_blk_cnt,
+			  vfc->vfc_large_thresh);
+		d_binheap_remove(&vfc->vfc_heap, &entry->ve_node);
+		entry->ve_in_heap = 0;
+	}
+	d_list_del_init(&entry->ve_link);
```

Comment:
We only need the list_del for non in_heap case, right?

Reply:
right.

Reply:
Let's add some list empty assertion here in later cleanup patch, what do you think?

Reply:
OK.